src/server/failover.c

```diff
@@ -1138,6 +1266,17 @@ be_secondary(time_t delay)
 					}
 					break;
 				}
+				/* Invoke stonith, to make sure primary is down */
+				rc = check_and_invoke_stonith(pbs_conf.pbs_primary);
+				if (rc) {
+					Secondary_state = SECONDARY_STATE_takeov;
```

Comment:
Won't Secondary_state be SECONDARY_STATE_takeov?  You are in that case of the switch() statement.

Reply:
Was doing this to make sure it comes back to try to execute the script again if an error occurs, but the Secondary_state shouldn't change anywhere in this block except when connection with primary is re-established.

Have removed it.